merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinBindings.java

```diff
@@ -189,6 +192,23 @@ private void getResourceSamples(final Context ctx) {
       ctx.status(404).result(ResponseSerializers.serializeNoSuchPlanException(ex).toString());
     }
 }
+  private void getConstraintViolations(final Context ctx) {
+    try {
+      final var planId = parseJson(ctx.body(), hasuraPlanActionP).input().planId();
+
+      final var constraintViolations = this.simulationAction.getViolations(planId);
```

Comment:
These calls to `getViolations` and `serializeConstraintViolations` show up in an earlier commit than the definitions of those methods do, so there are a couple commits at the beginning of this PR that don't compile.

I tried reversing the order of all four commits in this PR, and they all built nicely, so if that's okay with you, I might request a reversal of the commit order.